### PR TITLE
Filter out duplicate alerts

### PR DESCRIPTION
**Pull request by capsterx**

### CODE REVIEW

File: sopel_modules/weather/wz.py [hunk 1]
```diff
@@ -114,7 +114,13 @@ def get_current(self, city, state, weather):
         )
         if 'alerts' in weather:
             result += " | Alerts: "
-            result += ', '.join([x['title'] + ' ' + self.__short(x['uri']) for x in weather['alerts']])
+            uris = set()
+            alerts = []
+            for x in weather['alerts']:
+                if x['uri'] not in uris:
+                    alerts.append(x)
+                    uris.add(x['uri'])
```
Comment (cacoyle) [inline sopel_modules/weather/wz.py]: Couldn't this just be like `uris = set([x['uri'] for x in weather['alerts']])` ?

```
>>> weather = {}
>>> weather['alerts'] = [ {'uri': 'foo'}, {'uri': 'bar'}, {'uri': 'bin'}, {'uri: 'foo'}, {'uri': 'foo'} ]
>>> set([x['uri'] for x in weather['alerts']])
{'foo', 'bar', 'bin'}
```

Comment (capsterx) [reply sopel_modules/weather/wz.py]: The set is just there to filter the alerts, the set is not used after the filtering is done.  The alerts array is what we care about, not the uris.

Comment (capsterx) [reply sopel_modules/weather/wz.py]: You could do something like
seen = set()
alerts = [x for x in weather['alerts'] if x['uri'] not in seen and not seen.add(x['uri'])]

which i just found, but it's a lil odd to have a comprehension have side effects, to me.

Comment (cacoyle) [reply sopel_modules/weather/wz.py]: Your call

**Comment by cacoyle** [comment]
Also, while you're in here, bump sopel in requirements to `==7.1.7`  so it stops reverting sopel to 6.x

**Pull request merged**